### PR TITLE
FLOW-402 - Create getPageSize function to make paging limits consistent

### PR DESCRIPTION
**Pull request by dshoulders**

### CODE REVIEW

File: js/services/component.ts [hunk 1]
```diff
@@ -334,3 +337,27 @@ export const onOutcome = (outcome: any, objectData: any[], flowKey: string): JQu
...
+        `paging.${model.componentType.toLowerCase()}`, flowKey,
+    );
+
+    const pageLimitSettingFromListFilter = pathOr(null, ['objectDataRequest', 'listFilter', 'limit'], model);
+
+    const limit = usePaginationAttribute
+        ? pageLimitFromAttributes // 1st priority
+        : (
+            pageLimitSettingForComponentType // 2nd priority
+            || pageLimitSettingFromListFilter // 3rd priority
```
Comment (TomFullalove) [inline js/services/component.ts]: I have a feeling 2nd and 3rd priority should be swapped here, and then we wouldn't have to get rid of the table default from the settings.
What do you think?
Currently with this, users can still run into the same problem in the ticket for all the other components.

Comment (dshoulders) [reply js/services/component.ts]: Yes, that's what this statement in the issue does suggest: "The UI shouldn't be substituting its own value here (or even defaulting to anything, unless there's no listFilter at all)". I'll switch those around.

File: js/services/component.ts [hunk 2]
```diff
@@ -334,3 +337,34 @@ export const onOutcome = (outcome: any, objectData: any[], flowKey: string): JQu
...
+    const pageLimitSettingForComponentType = Settings.flow(
+        `paging.${model.componentType.toLowerCase()}`, flowKey,
+    );
+
+    const pageLimitSettingFromListFilter = pathOr(null, ['objectDataRequest', 'listFilter', 'limit'], model);
+
+    const limit = usePaginationAttribute
+        ? pageLimitFromAttributes // 1st priority
+        : (
+            pageLimitSettingFromListFilter // 2rd priority
```
Comment (TomFullalove) [inline js/services/component.ts]: 2rd->2nd
3nd->3rd

File: js/services/component.ts [hunk 3]
```diff
@@ -334,3 +337,34 @@ export const onOutcome = (outcome: any, objectData: any[], flowKey: string): JQu
...
+
+export const getPageSize = (model, flowKey) => {
+
+    const pageLimitFromAttributes = pathOr(null, ['attributes', 'paginationSize'], model);
+    const pageLimitFromAttributesIsValid = !Number.isNaN(pageLimitFromAttributes);
+
+    const usePaginationAttribute =
+        pageLimitFromAttributesIsValid &&
+        (
+            // Data is comming from a service, we can ignore "pagination" boolean attribute
```
Comment (TomFullalove) [inline js/services/component.ts]: comming-> coming

File: js/services/component.ts [hunk 4]
```diff
@@ -334,3 +337,34 @@ export const onOutcome = (outcome: any, objectData: any[], flowKey: string): JQu
...
+            !Utils.isNullOrUndefined(model.objectDataRequest)
+            // Data is comming from a list value, we need to check that the "pagination" attribute is also set to true
```
Comment (TomFullalove) [inline js/services/component.ts]: comming coming

**Comment by dshoulders** [comment]
> t

The function in ajax.ts is publicly accessible so we shouldn't change the functionality there.

**Comment by dshoulders** [comment]
> There is some limit logic in ajax.ts for files component.
> line: 37
> Would be good to see if we can absorb that into the new getPageSize function.
> But at first glance it doesn't look at if model/component are passed in...

The function in ajax.ts is publicly accessible so we shouldn't change the functionality there.

**Pull request merged**